probe/endpoint/resolver.go

```diff
@@ -44,41 +63,64 @@ func NewReverseResolver() *ReverseResolver {
 // gcache.NotFoundKeyError error otherwise. Note: it returns one of the
 // possible names that can be obtained for that IP.
 func (r *ReverseResolver) Get(address string) (string, error) {
-	val, err := r.cache.Get(address)
-	if hostname, ok := val.(string); err == nil && ok {
-		return hostname, nil
-	}
-	if _, ok := val.(struct{}); err == nil && ok {
-		return "", errNotFound
-	}
-	if err == gcache.NotFoundKeyError {
+	val, found := r.cache.Get(address)
+	if !found {
 		// We trigger a asynchronous reverse resolution when not cached
 		select {
 		case r.addresses <- address:
 		default:
 		}
+		return "", errNotFound
+	}
+	if _, ok := val.(struct{}); ok {
+		return "", errNotFound
 	}
-	return "", errNotFound
+	entry, ok := val.(*resolverCacheEntry)
+	if !ok {
+		panic(fmt.Sprintf("unknown entry type in cache: %+v", val))
```

Comment:
https://golang.org/ref/spec#Type_assertions

This is a programmer error if it fails, so I would just let the runtime panic (instead of checking ok... and panicing anyway).